### ISSUE TITLE
Timeout in more than thousand records

### ISSUE THREAD

**Issue by N3TC4T** [Knotis/djangocassandra#79]
Hi man 
It's a little bit wired , in more than thousand records i get timeout or it's take to many time to retirive records , but from django shell and limit it's working like a charm 

list(model_name.objects.all()[:20]

I think something in pagenation is not working well

**Comment by sethdenner** [comment]
What version are you running? I recently made some performance improvements
to the backend that solved a problem we were having with timeouts.
On May 29, 2016 01:41, "NETCAT" <notifications@github.com> wrote:

Hi man
It's a little bit wired , in more than thousand records i get timeout or
it's take to many time to retirive records , but from django shell and
limit it's working like a charm

list(model_name.objects.all()[:20]

I think something in pagenation is not working well

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Knotis/djangocassandra/issues/79>, or mute the thread
<https://github.com/notifications/unsubscribe/ABpt-3JIj2-_cuErZuoEdKsSQ2NVDD9Qks5qGVFAgaJpZM4IpPlV>
.

**Comment by N3TC4T** [comment]
> What version are you running?
thanks man for answer I'm running Version: 0.3.4

**Comment by N3TC4T** [comment]
After Update to 0.5.0 it's happened again , I get timeout

**Comment by sethdenner** [comment]
yeah count isn't a good operation to do with CQL. It's recommended that you use Counter Column Families. I want to implement automatic creation of a counter column families for facilitating the count method but I have not implemented this yet.

If you provide me with a specific use case I might be able to suggest a work around.

**Comment by N3TC4T** [comment]
I've check the cassandra cqlengine and in this file

> cassandra/cqlengine/statements.py LINE 585

they just use COUNT(*)

`    def __unicode__(self):
        qs = ['SELECT']
        if self.count:
            qs += ['COUNT(*)']`

so the fastest soloution is modfiying this but it's not a good solution

**Comment by N3TC4T** [comment]
Any Contact way ? So we can discuss about this ?

**Comment by N3TC4T** [comment]
For handleing the pagination and count problem the best soloution is using this pagination module 
https://github.com/ross/performant-pagination
